src/index.tsx

```diff
@@ -1,14 +1,16 @@
-import React, { createContext, useContext, useReducer, useEffect } from "react";
+import * as React from "react";
 
 import useLocalStorage from "./useLocalStorage";
 
+const { createContext, useContext, useReducer, useEffect } = React;
```

Comment:
```suggestion
const { createContext, useContext, useReducer, useEffect } = React;
```

I'd go as far as removing this and then invoking `React.createContext` if this is the style we want to adopt.

Reply:
We can turn on `--allowSyntheticDefaultImports`
https://www.typescriptlang.org/docs/handbook/compiler-options.html

If you want to use the original style.

Typescript doesn't allow us to import react as a default export because it is already built. That's why I opted for that style instead, but we can add `"allowSyntheticDefaultImports": true,` to allow for it, and keep the original style.

Reply:
Added allowSyntheticDefaultImports": true,

Reply:
I don't mind. My personal style these days is `import * as React from 'react'`. The React team seem to advocate this as well from what I've seen.

Reply:
I'll make it `* as React`, and then `React.functions`